pybop/costs/error_costs.py

```diff
@@ -38,7 +38,14 @@ def __call__(self, x, grad=None):
         Computes the cost.
         """
         try:
-            return np.sqrt(np.mean((self.problem.evaluate(x) - self._target) ** 2))
+            prediction = self.problem.evaluate(x)
+
+            if len(prediction) < len(self._target):
```

Comment:
We can return within the if statement:
```suggestion
            if len(prediction) < len(self._target):
                return np.Infinity  # simulation stopped early
            else:
                return p.sqrt(np.mean((prediction - self._target) ** 2))
```